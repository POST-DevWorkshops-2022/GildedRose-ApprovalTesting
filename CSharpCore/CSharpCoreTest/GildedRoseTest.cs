using Xunit;
using System.Collections.Generic;
using FluentAssertions;
using ApprovalTests.Reporters;
using ApprovalTests;
using Newtonsoft.Json;
using ApprovalTests.Combinations;

namespace CSharpCore.Test
{
    [UseReporter(typeof(VisualStudioReporter))]
    public class GildedRoseTest
    {
        [Fact]
        public void ShouldHaveItemWithExpectedName()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            Items[0].Name.Should().Be("foo");
        }

        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void VerifyStandardItemByToString()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            Approvals.Verify(Items[0].ToString());
        }

        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void VerifyStandardItemWithJson()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            Approvals.Verify(JsonConvert.SerializeObject(Items[0]));
        }

        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void VerifyStandardJsonItem()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            Approvals.VerifyJson(JsonConvert.SerializeObject(Items[0]));
        }

        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void VerifyAllItemCombinations()
        {
            string[] names = { "foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros", };
            int[] quality = { 0, 1, 49, 50 };
            int[] sellIn = { -1, 0, 2, 6, 11 };

            CombinationApprovals.VerifyAllCombinations(doUpdateQuality, names, sellIn, quality);
        }

        private string doUpdateQuality(string name, int sellIn, int quality)
        {
            Item[] items = new Item[] { new Item { Name = name, SellIn = sellIn, Quality = quality } };
            GildedRose app = new GildedRose(items);
            app.UpdateQuality();
            return JsonConvert.SerializeObject(items[0]);
        }
    }
}
