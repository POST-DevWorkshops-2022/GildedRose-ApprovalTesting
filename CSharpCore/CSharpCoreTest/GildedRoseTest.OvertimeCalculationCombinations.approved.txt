[foo,-1,0] => {"Name":"foo","SellIn":-2,"Quality":0}
[foo,-1,1] => {"Name":"foo","SellIn":-2,"Quality":0}
[foo,-1,49] => {"Name":"foo","SellIn":-2,"Quality":47}
[foo,-1,50] => {"Name":"foo","SellIn":-2,"Quality":48}
[foo,0,0] => {"Name":"foo","SellIn":-1,"Quality":0}
[foo,0,1] => {"Name":"foo","SellIn":-1,"Quality":0}
[foo,0,49] => {"Name":"foo","SellIn":-1,"Quality":47}
[foo,0,50] => {"Name":"foo","SellIn":-1,"Quality":48}
[foo,2,0] => {"Name":"foo","SellIn":1,"Quality":0}
[foo,2,1] => {"Name":"foo","SellIn":1,"Quality":0}
[foo,2,49] => {"Name":"foo","SellIn":1,"Quality":48}
[foo,2,50] => {"Name":"foo","SellIn":1,"Quality":49}
[foo,6,0] => {"Name":"foo","SellIn":5,"Quality":0}
[foo,6,1] => {"Name":"foo","SellIn":5,"Quality":0}
[foo,6,49] => {"Name":"foo","SellIn":5,"Quality":48}
[foo,6,50] => {"Name":"foo","SellIn":5,"Quality":49}
[foo,11,0] => {"Name":"foo","SellIn":10,"Quality":0}
[foo,11,1] => {"Name":"foo","SellIn":10,"Quality":0}
[foo,11,49] => {"Name":"foo","SellIn":10,"Quality":48}
[foo,11,50] => {"Name":"foo","SellIn":10,"Quality":49}
[Aged Brie,-1,0] => {"Name":"Aged Brie","SellIn":-2,"Quality":2}
[Aged Brie,-1,1] => {"Name":"Aged Brie","SellIn":-2,"Quality":3}
[Aged Brie,-1,49] => {"Name":"Aged Brie","SellIn":-2,"Quality":50}
[Aged Brie,-1,50] => {"Name":"Aged Brie","SellIn":-2,"Quality":50}
[Aged Brie,0,0] => {"Name":"Aged Brie","SellIn":-1,"Quality":2}
[Aged Brie,0,1] => {"Name":"Aged Brie","SellIn":-1,"Quality":3}
[Aged Brie,0,49] => {"Name":"Aged Brie","SellIn":-1,"Quality":50}
[Aged Brie,0,50] => {"Name":"Aged Brie","SellIn":-1,"Quality":50}
[Aged Brie,2,0] => {"Name":"Aged Brie","SellIn":1,"Quality":1}
[Aged Brie,2,1] => {"Name":"Aged Brie","SellIn":1,"Quality":2}
[Aged Brie,2,49] => {"Name":"Aged Brie","SellIn":1,"Quality":50}
[Aged Brie,2,50] => {"Name":"Aged Brie","SellIn":1,"Quality":50}
[Aged Brie,6,0] => {"Name":"Aged Brie","SellIn":5,"Quality":1}
[Aged Brie,6,1] => {"Name":"Aged Brie","SellIn":5,"Quality":2}
[Aged Brie,6,49] => {"Name":"Aged Brie","SellIn":5,"Quality":50}
[Aged Brie,6,50] => {"Name":"Aged Brie","SellIn":5,"Quality":50}
[Aged Brie,11,0] => {"Name":"Aged Brie","SellIn":10,"Quality":1}
[Aged Brie,11,1] => {"Name":"Aged Brie","SellIn":10,"Quality":2}
[Aged Brie,11,49] => {"Name":"Aged Brie","SellIn":10,"Quality":50}
[Aged Brie,11,50] => {"Name":"Aged Brie","SellIn":10,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,-1,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,-1,1] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,-1,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,-1,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,1] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,2,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":1,"Quality":3}
[Backstage passes to a TAFKAL80ETC concert,2,1] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":1,"Quality":4}
[Backstage passes to a TAFKAL80ETC concert,2,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":1,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,2,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":1,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,6,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":2}
[Backstage passes to a TAFKAL80ETC concert,6,1] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":3}
[Backstage passes to a TAFKAL80ETC concert,6,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,6,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,11,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":1}
[Backstage passes to a TAFKAL80ETC concert,11,1] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":2}
[Backstage passes to a TAFKAL80ETC concert,11,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,11,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":50}
[Sulfuras, Hand of Ragnaros,-1,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":0}
[Sulfuras, Hand of Ragnaros,-1,1] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":1}
[Sulfuras, Hand of Ragnaros,-1,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":49}
[Sulfuras, Hand of Ragnaros,-1,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":50}
[Sulfuras, Hand of Ragnaros,0,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":0}
[Sulfuras, Hand of Ragnaros,0,1] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":1}
[Sulfuras, Hand of Ragnaros,0,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":49}
[Sulfuras, Hand of Ragnaros,0,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":50}
[Sulfuras, Hand of Ragnaros,2,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":2,"Quality":0}
[Sulfuras, Hand of Ragnaros,2,1] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":2,"Quality":1}
[Sulfuras, Hand of Ragnaros,2,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":2,"Quality":49}
[Sulfuras, Hand of Ragnaros,2,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":2,"Quality":50}
[Sulfuras, Hand of Ragnaros,6,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":0}
[Sulfuras, Hand of Ragnaros,6,1] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":1}
[Sulfuras, Hand of Ragnaros,6,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":49}
[Sulfuras, Hand of Ragnaros,6,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":50}
[Sulfuras, Hand of Ragnaros,11,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":0}
[Sulfuras, Hand of Ragnaros,11,1] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":1}
[Sulfuras, Hand of Ragnaros,11,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":49}
[Sulfuras, Hand of Ragnaros,11,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":50}
